pandas/conftest.py

```diff
@@ -51,7 +50,6 @@ def add_imports(doctest_namespace):
 
 @pytest.fixture(params=['bsr', 'coo', 'csc', 'csr', 'dia', 'dok', 'lil'])
 def spmatrix(request):
-    tm._skip_if_no_scipy()
```

Comment:
Why can we just remove this?

Reply:
Instead of calling it imperatively I replaced it with decorators in the test functions that were using this fixture

Reply:
Hmm...I think that we should still skip within the resource if it cannot be used.  It is more decoupled that way IMO.

Reply:
So you'd rather remove the decorator and keep it to the fixture to skip tests? Or do both and live with the duplication? I'd argue either one violates TMTOWTDI given the former would make this inconsistent with other SciPy-requiring tests that don't need this fixture (which are the majority). Not to mention it would defeat the purpose of the original issue. The latter violation should be self-evident.

Reply:
yeah this is ok

Reply:
Fair enough.  You could just do `scipy = pytest.importorskip(...)`, but if we're okay with leaving it to the developer to be mindful of that dependency, sure.